accounting/account/models.py

```diff
@@ -0,0 +1,155 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+Account models used for user accounts in the Accounting service.
+"""
+
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from django_countries.fields import CountryField
+from djmoney.models.fields import MoneyField
+
+USER_MODEL = get_user_model()
+
+
+class Address(models.Model):
+    """
+    An address holding generic locational information.
+    """
+
+    country = CountryField(
+        help_text=_("The country associated with this user account."))
+    address_line1 = models.CharField(
+        max_length=128,
+        help_text=_("The first address line used to appear on accounting documents, i.e. invoices."))
+    address_line2 = models.CharField(
+        max_length=128, blank=True, null=True,
+        help_text=_("Additional line for extending an address."))
+    zipcode = models.CharField(
+        max_length=10,
+        help_text=_("A 5-digit or ZIP+4 zipcode. Example: 12345, 12345-6789."))
+    city = models.CharField(
+        max_length=60,
+        help_text=_("The city associated with this user account."))
+    state = models.CharField(
+        max_length=80, blank=True, null=True,
+        help_text=_("The state or province associated with this user account. "
+                    "Required if country is US, CA, AU, or BR."))
+
+    class Meta:
+        verbose_name = _('Address')
+        verbose_name_plural = _('Addresses')
+
+    def __str__(self):
+        """
+        Return a string identifying this address object.
+        """
+        # TODO: Handle the case where non-required values are missing, like `state` and `address_line2`.
+        return '{} {}, {} {} {}, {}'.format(
+            self.address_line1, self.address_line2, self.city, self.state, self.zipcode, self.country
+        )
+
+
+class Account(models.Model):
+    """
+    An account contains accounting details for a user, i.e. bank account details.
+
+    A user's accounts can have multiples of certain types of data.
+    For example, a user can have multiple bank accounts, multiple addresses, or multiple hourly rates
+    which differ per client where the user is the provider.
+    """
+
+    user = models.OneToOneField(
+        USER_MODEL, models.CASCADE, related_name='account',
+        help_text=_("The authentication user linked to this account."))
+    address = models.ForeignKey(
+        Address, models.CASCADE, related_name='accounts',
+        help_text=_("The address of this user."))
+    business_name = models.CharField(
+        max_length=120, blank=True, null=True,
+        help_text=_("The name of the business or company associated with this account."))
+    occupation = models.CharField(
+        max_length=80, blank=True, null=True,
+        help_text=_("The occupation of the user of this account. "
+                    "Required if country is US, CA, or JP."))
```

Comment:
This comes from Transferwise? It would be good to have a way to not have to update our code every time a requirement from transferwise changes - ie ask for this information dynamically, based on what they send through the API.

Reply:
I don't understand what you mean here -- the TransferWise API expects this information in a request to, for example, create a new recipient account. I guess you're saying we'd create an account on their side manually? I imagined we'd let that happen through the registration process -- user would enter these details and their recipient account would be made automatically if they are eligible for TransferWise payments.

Reply:
As mentioned elsewhere, for things that Transferwise require, this would be dynamic fields and likely only needs to be stored in Transferwise. If we were to store it on our side eventually, that would go to the JSON field.

Reply:
Just to be sure -- are you saying that the business name / occupation should be in a JSONField?